deps/sketch.py

```diff
@@ -0,0 +1,271 @@
+"""Sketch and library management"""
+
+from contextlib import asynccontextmanager
+import asyncio
+import io
+import json
+import os
+import re
+import zipfile
+from os import path
+
+import aiofiles
+import httpx
+from fastapi import FastAPI, HTTPException
+
+from deps.utils import repeat_every, check_for_internet
+from deps.logs import logger
+from models import Library, Sketch
+from conf import settings
+
+CWD = os.getcwd()
+
+fqbn_to_board = {  # Mapping from fqbn to PlatformIO board
+    "arduino:avr:uno": "uno",
+    "arduino:avr:nano": "nanoatmega328",
+    "arduino:avr:mega": "megaADK",
+    "arduino:esp32:nano_nora": "arduino_nano_esp32",
+}
+
+fqbn_to_platform = {  # Mapping from fqbn to PlatformIO platform
+    "arduino:avr:uno": "atmelavr",
+    "arduino:avr:nano": "atmelavr",
+    "arduino:avr:mega": "atmelavr",
+    "arduino:esp32:nano_nora": "espressif32",
+}
+
+library_index_json = {}
+library_indexed_json = {}
+
+
+def _get_latest_version(versions: list[str]) -> str:
+    """Return the latest version from a list of semantically versioned strings"""
+    return max(versions, key=lambda x: tuple(map(int, x.split("."))))
+
+
+def _parse_library_properties(properties: str) -> dict[str, str]:
+    """Parse a library.properties file and return the key-value pairs"""
+    return dict(line.split("=") for line in properties.split("\n") if "=" in line)
+
+
+async def _install_library_zip(
```

Comment:
why not use pio pkg install instead of this manual stuff? And then link the lib with something like this:
```
[env:myenv]
lib_deps =
    FooLib=symlink://../../shared-libraries/FooLib 
```